analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -423,18 +424,18 @@ class SpdxDocumentFile(
             )
         }
 
-    override fun mapDefinitionFiles(definitionFiles: List<File>) =
-        definitionFiles.filter { definitionFile ->
-            val spdxDocument = SpdxModelMapper.read<SpdxDocument>(definitionFile)
-
+    override fun mapDefinitionFiles(definitionFiles: List<File>): List<File> =
+        definitionFiles.associateWith {
+            SpdxModelMapper.read<SpdxDocument>(it)
+        }.filterTo(spdxDocumentForFile) { (_, spdxDocument) ->
             // Distinguish whether we have a project-style SPDX document that describes a project and its dependencies,
             // or a package-style SPDX document that describes a single (dependency-)package.
             spdxDocument.isProject()
-        }
+        }.keys.toList()
 
     override fun resolveDependencies(definitionFile: File): List<ProjectAnalyzerResult> {
         val workingDir = definitionFile.parentFile
-        val spdxDocument = SpdxModelMapper.read<SpdxDocument>(definitionFile)
+        val spdxDocument = spdxDocumentForFile.getValue(definitionFile)
```

Comment:
Would it make sense to fallback to `SpdxModelMapper.read<SpdxDocument>(it)` ?

Reply:
That exception actually should never occur, as `resolveDependencies()` does not see any files that have not seen by `mapDefinitionFiles()` before. So no, such a fallback should not be necessary.

Reply:
Not sure what is going on, but it seems that this exception actually occurred in the tests.

Reply:
How about extracting a function like
```
fun readSpdxDocument(file: File): SpdxDocument =
    spdxDocumentForFile.getOrPut(file) { SpdxModelMapper.read<SpdxDocument>(file) } 
```

Then there is no assumption introduced that mapDefinitionFile is called first.

Reply:
A test called `resolveDependencies()` directly, bypassing `mapDefinitionFiles()`, so I ended up adding the fallback that @fviernau suggested.